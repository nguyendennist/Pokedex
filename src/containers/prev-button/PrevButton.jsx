import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from 'components/button/Button.jsx';
import decrementOffset from 'actions/pokemon/decrement-offset/decrement-offset';

const PrevButton = props => {
	const prevPage = () => {
		if (props.pokemons.length === 20) {
			props.handleOffsetChange();
			setTimeout(() => props.fetchPokemons(), 250);
		}
	};

	return (<Button className='prev-button' name='prev' onClick={prevPage} />);
};

PrevButton.propTypes = {
	fetchPokemons: PropTypes.func.isRequired,
	handleOffsetChange: PropTypes.func.isRequired,
	pokemons: PropTypes.array.isRequired
};

const mapDispatchToProps = dispatch => ({
	handleOffsetChange: () => {
		dispatch(decrementOffset(20));
	}
});

const mapStateToProps = state => ({
	pokemons: state.pokemons
});

export default connect(mapStateToProps, mapDispatchToProps)(PrevButton);
