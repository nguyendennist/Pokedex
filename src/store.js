import { createStore } from 'redux';
import pokemon from 'reducers/pokemon/pokemon';

const store = createStore(pokemon);

export default store;
