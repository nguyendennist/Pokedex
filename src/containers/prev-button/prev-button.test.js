import React from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import store from 'store';
import PrevButton from './PrevButton.jsx';

describe('unit tests', () => {
	test('should be defined', () => {
		expect(PrevButton).toBeDefined();
	});
});

describe('ui tests', () => {
	test('should render correctly', () => {
		const tree = create(
			<Provider store={store}>
				<PrevButton />
			</Provider>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
