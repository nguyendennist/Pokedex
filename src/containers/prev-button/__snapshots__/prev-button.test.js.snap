// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui tests should render correctly 1`] = `
<div
  className="prev-button"
  onClick={[Function]}
>
  <p>
    prev
  </p>
</div>
`;
